libs/core/src/main/java/org/opensearch/core/util/BytesRefUtils.java

```diff
@@ -103,12 +104,12 @@ public static int sortAndDedup(final BytesRefArray bytes, final int[] indices) {
         return uniqueCount;
     }
 
+    /**
+     * Decodes a long value written as bytes in big endian order.
+     * @param bytes in big endian order
+     * @return long value
+     */
     public static long bytesToLong(BytesRef bytes) {
-        int high = (bytes.bytes[bytes.offset + 0] << 24) | ((bytes.bytes[bytes.offset + 1] & 0xff) << 16) | ((bytes.bytes[bytes.offset + 2]
-            & 0xff) << 8) | (bytes.bytes[bytes.offset + 3] & 0xff);
-        int low = (bytes.bytes[bytes.offset + 4] << 24) | ((bytes.bytes[bytes.offset + 5] & 0xff) << 16) | ((bytes.bytes[bytes.offset + 6]
-            & 0xff) << 8) | (bytes.bytes[bytes.offset + 7] & 0xff);
-        return (((long) high) << 32) | (low & 0x0ffffffffL);
+        return (long) BitUtil.VH_BE_LONG.get(bytes.bytes, 0);
```

Comment:
I think this is not right, the `bytes.offset` is not taken into account:

```suggestion
        return (long) BitUtil.VH_BE_LONG.get(bytes.bytes, bytes.offset);
```

Reply:
Ah, good catch! Fixed and updated the unit-test to catch this.